src/Interpreters/ActionsDAG.cpp

```diff
@@ -836,7 +842,12 @@ ActionsDAGPtr ActionsDAG::foldActionsByProjection(const std::unordered_map<const
     }
 
     for (const auto * output : required_outputs)
-        dag->outputs.push_back(mapping[output]);
+    {
+        const auto * mapped_output = mapping[output];
```

Comment:
Keep the names for outputs.
Add an alias if the mapped node has a different result name.